### PR TITLE
[Feature][SQL Config] Add SQL config adapter

### PR DESCRIPTION
**Pull request by rewerma**
### Purpose of this pull request
[Feature][SQL Config] Add SQL config adapter

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  'driver' = 'com.mysql.cj.jdbc.Driver',
+  'user' = 'root',
+  'password' = 'Abc!@#135_seatunnel',
+  'connection_check_timeout_sec' = '100',
+  'generate_sink_sql' = 'true',
+  'database' = 'seatunnel',
+  'table' = 'sink',
+  'result_table_name' = 'temp1',
+  'type'='sink'
+);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel.sql]: missing `INSERT INTO ...` ？

Comment (rewerma) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel.sql]: Yes, using the 'source_table_name' configuration option in the sink statement allows avoiding the insert into select statement.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel.sql]: I suggest you force users to assemble the source/sink relationship themselves using sql

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink_parallel.sql]: 1. Avoid confusing relationships when there are multiple sources/sinks
2. Using `INSERT INTO ... SELECT ...` users can filter/transform fields by themselves

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+CREATE TABLE temp1 AS
+    SELECT c_bit_1, c_bit_8, c_bit_16, c_bit_32, c_bit_64, c_boolean, c_tinyint, c_tinyint_unsigned, c_smallint, c_smallint_unsigned,
+           c_mediumint, c_mediumint_unsigned, c_int, c_integer, c_bigint, c_bigint_unsigned,
+           c_decimal, c_decimal_unsigned, c_float, c_float_unsigned, c_double, c_double_unsigned,
+           c_char, c_tinytext, c_mediumtext, c_text, c_varchar, c_json, c_longtext, c_date,
+           c_datetime, c_time, c_timestamp, c_tinyblob, c_mediumblob, c_blob, c_longblob, c_varbinary,
+           c_binary, c_year, c_int_unsigned, c_integer_unsigned,c_bigint_30,c_decimal_unsigned_30,c_decimal_30 FROM source_table;
+
+INSERT INTO sink_table SELECT * FROM temp1;
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink.sql]: Is it possible to directly use `INSERT INTO sink_table SELECT a as a1, b, c as c1,... FROM source_table`

Comment (rewerma) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-1/src/test/resources/jdbc_mysql_source_and_sink.sql]: Yes, support this syntax.

File: docs/en/concept/sql-config.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+```sql
+/* config
+env {
+  parallelism = 1
+  job.mode = "BATCH"
+}
+*/
+
+CREATE TABLE source_table WITH (
+  'connector'='jdbc',
```
Comment (hailin0) [inline docs/en/concept/sql-config.md:18]: Is it better for this field value refer to [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties)?

File: docs/zh/concept/sql-config.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  'url' = 'jdbc:mysql://localhost:3306/seatunnel',
+  'driver' = 'com.mysql.cj.jdbc.Driver',
+  'user' = 'root',
+  'password' = '123456',
+  'query' = 'select * from source',
+  'properties' = '{
+    useSSL = false,
+    rewriteBatchedStatements = true
+  }',
+  'type'='source'
```
Comment (hailin0) [inline docs/zh/concept/sql-config.md]: move `type` to `connector` after

Comment (rewerma) [reply docs/zh/concept/sql-config.md]: ok

**Comment by hailin0** [comment]
cc @EricJoy2048 @Hisoka-X

File: docs/en/concept/sql-config.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+* Note: This syntax does **not support** specifying fields in `INSERT`, like this: `INSERT INTO sink_table (id, name, age, email) SELECT id, name, age, email FROM source_table;`
+
+### INSERT INTO SELECT TABLE Syntax
+
+```sql
+INSERT INTO sink_table SELECT source_table;
+```
+
+* The `SELECT` part directly uses the name of the source-mapped table, indicating that all data from the source table will be inserted into the target table.
+* sing this syntax does not generate related `transform` configurations. This syntax is generally used in multi-table synchronization scenarios. For example:
```
Comment (dailai) [inline docs/en/concept/sql-config.md]: The first letter needs to be capitalized.

**Comment by dailai** [comment]
Could you add some unit test for the new module named seatunnel-config-sql to vertify the logical? I think you can add some case like that the password in options is auto generate like this: ^%QjiXCG&amp;GR'ob-T8Lr6UNaqQC#;7).

image

File: seatunnel-config/seatunnel-config-sql/src/main/java/org/apache/seatunnel/config/sql/SqlConfigBuilder.java [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+    private static SinkConfig parseSinkSql(Map<String, String> options) {
+        String connector = options.get(OPTION_TABLE_CONNECTOR_KEY);
+        if (StringUtils.isEmpty(connector)) {
+            throw new ParserException("The connector of option is none");
+        }
+        SinkConfig sinkConfig = new SinkConfig();
+        sinkConfig.setConnector(connector);
+        // original sink table without source_table_name
+        options.remove(OPTION_SOURCE_TABLE_NAME_KEY);
+        // String sourceTableName = options.get(OPTION_SOURCE_TABLE_NAME_KEY);
```
Comment (dailai) [inline seatunnel-config/seatunnel-config-sql/src/main/java/org/apache/seatunnel/config/sql/SqlConfigBuilder.java]: Please remove useless code.

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLType.java [hunk 7]
```diff
@@ -177,12 +177,26 @@ public SeaTunnelDataType<?> getExpressionType(Expression expression) {
             SeaTunnelDataType<?> leftType = getExpressionType(binaryExpression.getLeftExpression());
             SeaTunnelDataType<?> rightType =
                     getExpressionType(binaryExpression.getRightExpression());
-            if (leftType.getSqlType() == SqlType.INT && rightType.getSqlType() == SqlType.INT) {
+            if ((leftType.getSqlType() == SqlType.TINYINT
```
Comment (dailai) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLType.java:180]: Could you add  some test case for this to cover all the logical?

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLType.java [hunk 8]
```diff
@@ -177,12 +177,26 @@ public SeaTunnelDataType<?> getExpressionType(Expression expression) {
...
+                            || leftType.getSqlType() == SqlType.INT)
+                    && (rightType.getSqlType() == SqlType.TINYINT
+                            || rightType.getSqlType() == SqlType.SMALLINT
+                            || rightType.getSqlType() == SqlType.INT)) {
                 return BasicType.INT_TYPE;
             }
-            if ((leftType.getSqlType() == SqlType.INT || leftType.getSqlType() == SqlType.BIGINT)
-                    && (rightType.getSqlType() == SqlType.INT
-                            || rightType.getSqlType() == SqlType.BIGINT)) {
+            if ((leftType.getSqlType() == SqlType.TINYINT
```
Comment (dailai) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLType.java:188]: As above?

**Review by dailai — APPROVED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**